test/testunusedfunctions.cpp

```diff
@@ -769,6 +770,13 @@ class TestUnusedFunctions : public TestFixture {
               "}\n");
         ASSERT_EQUALS("[test.cpp:2]: (style) The function 'Type' is never used.\n", errout_str());
     }
+
+    void attributeCleanup()
+    {
+        check("void clean(void *ptr) {}\n"
+              "void * __attribute__((cleanup(clean))) p;\n");
+        ASSERT_EQUALS("", errout_str());
```

Comment:
The example compiles but results in
~~~
<source>:2:27: warning: 'cleanup' attribute only applies to local variables [-Wignored-attributes]
    2 |     void * __attribute__((cleanup(clean))) p;
~~~
But it probably doesn't matter for the test.